components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/extension/AbstractWorkflowRequestHandler.java

```diff
@@ -209,4 +214,23 @@ private boolean isWorkflowCompleted() {
             return true;
         } else return false;
     }
+
+    /**
+     * We can check whether the current event type is already associated with
+     * at-least one association or not by using isAssociated method.
+     *
+     * @return Boolean value for result of isAssociated
+     * @throws WorkflowException
+     */
+    public boolean isAssociated() throws WorkflowException{
```

Comment:
Yes, We will achieve that as improvement and this method will not return invalid result.